src/Native/Unix/System.Native/pal_interfaceaddresses.c

```diff
@@ -38,6 +44,42 @@
 #include <net/route.h>
 #endif
 
+static inline uint8_t mask2prefix(uint8_t * mask, int length)
+{
+    uint8_t len = 0;
+    uint8_t *end = mask + length;
+
+    if (mask == NULL)
+    {
+        // If we did not get valid mask, assume host address.
+        return (uint8_t)length * 8;
+    }
+
+    // Get whole bytes
+    while ((mask < end) && (*mask == 0xff))
+    {
+        len += 8;
+        mask ++;
+    }
+
+    // Get last incomplete byte
+    if (mask < end)
+    {
+        while (*mask)
```

Comment:
This is probably my not understanding the purpose of the function, but... what happens if mask is much less than end?  It's ok that we may not explore the whole thing?

Reply:
This is only called with NUM_BYTES_IN_IPV4_ADDRESS and NUM_BYTES_IN_IPV4_ADDRESS e.g. 4 and 16.